File: tests/human/x/php/cross_join_triple.php
<?php
$nums = [1,2];
$letters = ["A","B"];
$bools = [true,false];
$combos = [];
foreach ($nums as $n) {
    foreach ($letters as $l) {
        foreach ($bools as $b) {
            $combos[] = ["n"=>$n,"l"=>$l,"b"=>$b];
        }
    }
}
_print("--- Cross Join of three lists ---");
foreach ($combos as $c) {
    _print($c['n'], $c['l'], $c['b'] ? 'true' : 'false');
}

function _print(...$args) {
    $parts = [];
    foreach ($args as $a) {
        if (is_array($a) || is_object($a)) {
            $parts[] = json_encode($a);
        } else {
            $parts[] = strval($a);
        }
    }
    echo implode(' ', $parts), PHP_EOL;
}
?>

File: tests/transpiler/x/pl/cross_join_triple.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    Nums = [1, 2],
    Letters = ["A", "B"],
    Bools = [true, false],
    Combos = [_{n: 1, l: "A", b: true}, _{n: 1, l: "A", b: false}, _{n: 1, l: "B", b: true}, _{n: 1, l: "B", b: false}, _{n: 2, l: "A", b: true}, _{n: 2, l: "A", b: false}, _{n: 2, l: "B", b: true}, _{n: 2, l: "B", b: false}],
    writeln("--- Cross Join of three lists ---"),
    C = _{n: 1, l: "A", b: true},
    writeln("1 A true"),
    C1 = _{n: 1, l: "A", b: false},
    writeln("1 A false"),
    C2 = _{n: 1, l: "B", b: true},
    writeln("1 B true"),
    C3 = _{n: 1, l: "B", b: false},
    writeln("1 B false"),
    C4 = _{n: 2, l: "A", b: true},
    writeln("2 A true"),
    C5 = _{n: 2, l: "A", b: false},
    writeln("2 A false"),
    C6 = _{n: 2, l: "B", b: true},
    writeln("2 B true"),
    C7 = _{n: 2, l: "B", b: false},
    writeln("2 B false").
